### PR TITLE
bpo-44929: Fix enum string representation in the REPL

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue44929

### CODE REVIEW

**Comment by ambv** [comment]
The entire `global_flag_repr` function is wrong and untested. I added a few failing tests for now, I'm not sure how they should be fixed since I can't figure out what the intent is from the docs.

File: Lib/enum.py [hunk 1]
```diff
@@ -1396,7 +1396,7 @@ def global_flag_repr(self):
     module = self.__class__.__module__
     cls_name = self.__class__.__name__
     if self._name_ is None:
-        return "%x" % (module, cls_name, self._value_)
+        return "%s.%s(%x)" % (module, cls_name, self._value_)
```
Comment (gpshead) [inline Lib/enum.py]: Do any of the unittests test this particular format codepath?  none of the added tests have reprs that end with `(0x8)` or similar.

Comment (ambv) [reply Lib/enum.py]: Yeah, no, I couldn't quite figure out how to make `_name_` to be None. When we hear back from Ethan, he'll hopefully suggest a test for this, too. Worst case we can just add a test using `repr(re.RegexFlag(0))`.

Comment (ethanfurman) [reply Lib/enum.py]: https://bugs.python.org/msg399863

**Review by ethanfurman — DISMISSED** [review]
Fixed tests and global reprs.  Thank you.

File: Lib/test/test_enum.py [hunk 2]
```diff
@@ -3224,6 +3244,26 @@ def test_repr(self):
...
+    def test_global_repr_conform1(self):
+        self.assertEqual(
+                repr(HeadlightsC(0)),
+                '%s.OFF_C' % SHORT_MODULE,
+                )
+        self.assertEqual(
+                repr(HeadlightsC(2**0 + 2**2 + 2**3)),
+                '%(m)s.LOW_BEAM_C|%(m)s.FOG_C' % {'m': SHORT_MODULE},
+                )
+
```
Comment (ethanfurman) [inline Lib/test/test_enum.py:3274]: Still need test for `_name_ is None` branch.  I'll have that in shortly.

**Review by ethanfurman — CHANGES_REQUESTED** [review]

**Pull request merged**